### ISSUE TITLE
404 on files with url-encoded names

### ISSUE THREAD

**Issue by wqweto** [gogits/gogs#1137]
When I commit a file with name `ifcfg-eth1%3A1` it is not accessible from the web interface.

`/{project}/src/master/{folder}/ifcfg-eth1%3A1` returns 404

`ifcfg-eth1%3A1` is the Windows name of `ifcfg-eth1:1` copied over sftp from a Linux box.

**Comment by Unknwon** [comment]
Hi, I've tested both file name `ifcfg-eth1:1` and `ifcfg-eth1%3A1` and commit, there seems no problem(404) and page looks normal.

Can you provide more details on this?

**Comment by wqweto** [comment]
I still get 404 with Application Version: 0.6.3.0725 Beta

These are last couple of lines from `gogs.log`
```
[Macaron] Started GET /wqweto/Test/src/master/aaa for 127.0.0.1
[Macaron] Completed /wqweto/Test/src/master/aaa 200 OK in 28.754599ms
[Macaron] Started GET /wqweto/Test/src/master/ifcfg-eth2%3A1 for 127.0.0.1
[Macaron] Completed /wqweto/Test/src/master/ifcfg-eth2%3A1 404 Not Found in 24.6563ms
2015/07/26 22:42:17 [1;31m[view.go:56 Home()] [E] GetTreeEntryByPath: GetTreeEntryByPath: error not exist[0m
```
I'm on a CentOS 6.6 with nginx in front in simple proxy_pass mode
```
  location / {
    proxy_pass      http://localhost:3000;
  }
```
404 is returned in both http status code and a pretty customized page from gogs.

**Comment by Unknwon** [comment]
Well, my question is how `ifcfg-eth2%3A1` file is generated?

**Comment by wqweto** [comment]
I made some progress, I was able to access the file by manually editing url to `/{user}/{project}/src/master/ifcfg-eth1%253A2` i.e. I had to url_encode the % in the file name.

The real file name is `ifcfg-eth2%3A1` on an NTFS volume (Windows box). This name has to be url_encoded to `ifcfg-eth1%253A2` in the actual link address when listing files in repo directory for the link to be correctly decoded w/o converting `%3A` to `:` because file `ifcfg-eth2:1` is not part of the repo and consequently `GetTreeEntryByPath` complains.
